### PR TITLE
Add new CSE error code (83) for GPU download failures: Nvidia driver and fabric manager downloads

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adding a separate error code for GPU download failures. Previously download and install failures had the same failure code. 

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
Added new error code: ERR_GPU_DOWNLOAD_TIMEOUT (83) for GPU download issues (Nvidia GPU driver and Fabric manager downloads) .
```

### CODE REVIEW

File: pkg/agent/testdata/AKSUbuntu1604+K8S117/line48.sh [hunk 1]
```diff
@@ -56,8 +56,8 @@ downloadGPUDrivers() {
     fi
 
     mkdir -p ${GPU_DEST}
-    retrycmd_if_failure 30 5 3600 apt-get install -y linux-headers-$(uname -r) gcc make dkms || exit $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
-    retrycmd_if_failure 30 5 60 curl -fLS https://us.download.nvidia.com/tesla/$GPU_DV/NVIDIA-Linux-x86_64-${GPU_DV}.run -o ${GPU_DEST}/nvidia-drivers-${GPU_DV} || exit $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
+    retrycmd_if_failure 30 5 3600 apt-get install -y linux-headers-$(uname -r) gcc make dkms || exit $ERR_GPU_DRIVERS_DOWNLOAD_TIMEOUT
```
Comment (ganeshkumar5699) [inline pkg/agent/testdata/AKSUbuntu1604+K8S117/line48.sh:59]: I replaced `ERR_GPU_DRIVERS_INSTALL_TIMEOUT` with the new `ERR_GPU_DRIVERS_DOWNLOAD_TIMEOUT` here, as it is in the `downloadGPUDrivers` function. 

Would `ERR_APT_INSTALL_TIMEOUT` or another error code be more appropriate?

Comment (alexeldeib) [reply pkg/agent/testdata/AKSUbuntu1604+K8S117/line48.sh:59]: sgtm

**Comment by alexeldeib** [comment]
probably one more error code during VHD build as well?

**Review by alexeldeib — APPROVED** [review]
minor comments

File: pkg/agent/testdata/AKSUbuntu1604+K8S117/line48.sh (hunk 1 above)
Comment (ganeshkumar5699) [reply pkg/agent/testdata/AKSUbuntu1604+K8S117/line48.sh]: Replaced this with `ERR_APT_INSTALL_TIMEOUT`.

**Comment by ganeshkumar5699** [comment]
> probably one more error code during VHD build as well?

Added `ERR_GPU_DOWNLOAD_TIMEOUT` as error for Fabric manager download in install-deps, like you suggested. So there's a GPU download error in CSE and one in install-deps now.

**Pull request merged**